lib/ansible/modules/network/fortios/fortios_dlp_settings.py

```diff
@@ -0,0 +1,271 @@
+  returned: always
+  type: str
+  sample: "webfilter"
+revision:
+  description: Internal revision number
+  returned: always
+  type: str
+  sample: "17.0.2.10658"
+serial:
+  description: Serial number of the unit
+  returned: always
+  type: str
+  sample: "FGVMEVYYQT3AB5352"
+status:
+  description: Indication of the operation's result
+  returned: always
+  type: str
+  sample: "success"
+vdom:
+  description: Virtual domain used
+  returned: always
+  type: str
+  sample: "root"
+version:
+  description: Version of the FortiGate
+  returned: always
+  type: str
+  sample: "v5.6.3"
+
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+fos = None
+
+
+def login(data):
+    host = data['host']
+    username = data['username']
+    password = data['password']
+
+    fos.debug('on')
+    if 'https' in data and not data['https']:
+        fos.https('off')
+    else:
+        fos.https('on')
+
+    fos.login(host, username, password)
+
+
+def filter_dlp_settings_data(json):
+    option_list = ['cache-mem-percent', 'chunk-size', 'db-mode',
+                   'size', 'storage-device']
+    dictionary = {}
+
+    for attribute in option_list:
+        if attribute in json and json[attribute] is not None:
+            dictionary[attribute] = json[attribute]
+
+    return dictionary
+
+
+def dlp_settings(data, fos):
+    vdom = data['vdom']
+    dlp_settings_data = data['dlp_settings']
+    filtered_data = filter_dlp_settings_data(dlp_settings_data)
+    return fos.set('dlp',
+                   'settings',
+                   data=filtered_data,
+                   vdom=vdom)
+
+
+def fortios_dlp(data, fos):
+    login(data)
+
+    methodlist = ['dlp_settings']
+    for method in methodlist:
+        if data[method]:
+            resp = eval(method)(data, fos)
+            break
+
+    fos.logout()
+    return not resp['status'] == "success", resp['status'] == "success", resp
+
+
+def main():
+    fields = {
+        "host": {"required": True, "type": "str"},
+        "username": {"required": True, "type": "str"},
+        "password": {"required": False, "type": "str", "no_log": True},
+        "vdom": {"required": False, "type": "str", "default": "root"},
+        "https": {"required": False, "type": "bool", "default": "False"},
+        "dlp_settings": {
+            "required": False, "type": "dict",
+            "options": {
+                "cache-mem-percent": {"required": False, "type": "int"},
+                "chunk-size": {"required": False, "type": "int"},
+                "db-mode": {"required": False, "type": "str",
+                            "choices": ["stop-adding", "remove-modified-then-oldest", "remove-oldest"]},
+                "size": {"required": False, "type": "int"},
+                "storage-device": {"required": False, "type": "str"}
+
+            }
+        }
+    }
+
+    module = AnsibleModule(argument_spec=fields,
+                           supports_check_mode=False)
+    try:
```

Comment:
Move this to a common function which check this import like [this](https://github.com/ansible/ansible/blob/d252cc7a2bd05fad2c694727c715df222b3ed58f/lib/ansible/module_utils/vmware.py#L782)

Reply:
Hello @Akasurde ,

Thanks for the suggestion, will definitely look into it.
I do not see a need to rush this change and why it should block this code to move in.

Can we track this change here: https://github.com/fortinet-solutions-cse/ansible_generator/issues instead of blocking this proposal ?

Reply:
OK.

Reply:
Yup, I'm happy to accept a follow up bulk PR to update existing modules

Reply:
Hi @gundalow and @Akasurde ,
 I'm finally implementing this and other changes suggested. However I have two concerns that I'd like to raise here before doing all the changes:

1. fortiosapi is used in this module. If fortiosapi import is moved to a common util module then it will have to be referenced from here by using something like:
`fos = fortioshandler.fortiosapi.FortiOSAPI()`
 (assuming fortioshandler is the common util module). 
However I read in some places (e.g. stackoverflow) that this is not a good practice and instead the import should appear in the module where it is used. Can we still keep the import here?

2. Some users like to run the latest changes of our modules and for that they download our modules and run them from a different directory (the modules are not placed in Ansible official directory, instead they can be placed in user's home or somewhere else).E.g.: They download the code in a separated directory, do export ANSIBLE_LIBRARY=.../....../..... and run their playbooks.
My concern is that If this Ansible module fully relies on a common util module (e.g. fortioshandler) then the users will find it more difficult to run the latest modules in the same way that was done before. Probably doing export ANSIBLE_LIBRARY will not be enough. Is there a proper workaround for this problem?

Your opinion on this is very appreciated.
Thanks.